web/tests/typing_status.test.js

```diff
@@ -50,12 +50,12 @@ run_test("basics", ({override, override_rewire}) => {
     set_global("clearTimeout", clear_timeout);
 
     function notify_server_start(recipient) {
-        assert.deepStrictEqual(recipient, [1, 2]);
+        assert.deepStrictEqual(recipient, {message_type: "direct", ids: [1, 2]});
```

Comment:
For direct message `recipient ` is not an array anymore. So tests have to changed a little.